### ISSUE TITLE
[Question] Redirect from groups?

### ISSUE THREAD

**Issue by peterchoo** [meteorhacks/flow-router#182]
I've been looking at using groups to organise my routes, and one feature which appears to be missing from the groups is the ability to redirect to a route if the URL defined by the group is hit.

Consider the following pages:

- User list
- User edit
- Organisation edit

All of these pages are to be housed under the `/admin` section of the application.  Using just routes I have done the following:

```javascript
Router.route('/admin', {
  name: 'admin',
  action: function() {
    Router.go('admin.users');
  }
});

Router.route('/admin/users', {
  name: 'admin.users',
  ...
});

Router.route('/admin/users/:userId', {
  name: 'admin.users.edit',
  ...
});

Router.route('/admin/organisation', {
  name: 'admin.organisation',
  ...
});
```

Using a group, if I navigate to `/admin` I can't find a way of redirecting to `/admin/users`.

Is this a feature which could be added, or should I carry on defining my routes as I have been?

Thanks

**Comment by arunoda** [comment]
I think you can do the same with groups. Create a group for `/admin` and create `/` route inside to do the redirect. I think it should be `FlowRouter` not just `Router`.

~~~js
var adminRoutes = FlowRouter.group({
  prefix: '/admin'
});

adminRoutes.route('/', {
  triggersEnter: [function(contex, redirect) {
    redirect('/admin/users');
  }];
});

adminRoutes.route('/users', {
  action: function() {
    console.log("Yeah, I'm on /users");
  }
});
~~~

**Comment by peterchoo** [comment]
Hi,

Thanks for the reply, that will indeed work for me.  The `Router` vs `FlowRouter` was purely haste on my part.

I have a suggestion: Could we pass in `self.go` on both https://github.com/meteorhacks/flow-router/blob/02758cb1411ce2705c8729fee61d369e1f255134/client/router.js#L85 and #L96 ? This would allow us to pass in a route name as well as a url.

As another observation, can the "already redirected" Error be triggered?  Wouldn't the return on #L82 exit the runTriggers function?

**Comment by peterchoo** [comment]
I've had a little play:

Changing the `runTriggers` calls in `client/router.js` to use a bound call to `Router.go()` does let you use route names.

```javascript
    Triggers.runTriggers(
      triggers,
      self._current,
      _.bind(self.go, self),
      afterAllTriggersRan
    );
```


```javascript
Admin = FlowRouter.group({
  prefix: '/admin'
});

Admin.route('/', {
  triggersEnter: [function(context, redirect) {
    redirect('admin.users');
  }]
});
```

I haven't done any in depth testing, can you think of nay reason you wouldn't want to allow route names in this situation?

Thanks for your time.

**Comment by arunoda** [comment]
Okay. I got it. Let's see what we can do here.

**Comment by arunoda** [comment]
Implemented this and release with `1.17.2`

**Comment by peterchoo** [comment]
:+1: Thanks for your fast response! Updating now!
